test-suite/shared/src/test/scala/org/scalajs/testsuite/compiler/RuntimeTypeTestsTest.scala

```diff
@@ -0,0 +1,539 @@
+      test(expected, value, classOf[Long], _.isInstanceOf[Long], _.asInstanceOf[Long])
+    @inline def testFloat(expected: Boolean, value: Any): Unit =
+      test(expected, value, classOf[Float], _.isInstanceOf[Float], _.asInstanceOf[Float])
+    @inline def testDouble(expected: Boolean, value: Any): Unit =
+      test(expected, value, classOf[Double], _.isInstanceOf[Double], _.asInstanceOf[Double])
+
+    testBoolean(true, false)
+    testByte(true, 65.toByte)
+    testShort(true, 654.toShort)
+    testInt(true, -4321)
+    testLong(true, 684321L)
+    testFloat(true, 3.14f)
+    testDouble(true, 3.14)
+
+    testFloat(!executingInJVM, 45)
+    testDouble(!executingInJVM, 45)
+    testInt(!executingInJVM, 3.0f)
+    testDouble(!executingInJVM, 3.0f)
+    testInt(!executingInJVM, 5.0)
+    testFloat(!executingInJVM, 5.0)
+
+    testInt(!executingInJVM, 5.toByte)
+    testInt(!executingInJVM, 5.toShort)
+    testByte(!executingInJVM, 5)
+    testShort(!executingInJVM, 5)
+    testInt(!executingInJVM, 0.0)
+    testFloat(!executingInJVM, 0.0)
+    testFloat(!executingInJVM, -0.0)
+
+    testBoolean(false, 12345)
+    testChar(false, 12345)
+    testByte(false, 'a')
+    testShort(false, 'b')
+    testInt(false, 'c')
+    testLong(false, 'd')
+    testFloat(false, 'e')
+    testDouble(false, 'f')
+
+    // Special cases for negative 0, NaN and Infinity
+
+    testFloat(true, -0.0f)
+    testFloat(true, Float.NaN)
+    testFloat(true, Float.PositiveInfinity)
+
+    testDouble(true, -0.0)
+    testDouble(true, Double.NaN)
+    testDouble(true, Double.PositiveInfinity)
+
+    testInt(false, -0.0)
+    testInt(false, Double.NaN)
+    testInt(false, Double.PositiveInfinity)
+  }
+
+  @Test def unit(): Unit = {
+    import scala.runtime.BoxedUnit
+
+    @inline def testUnit(expected: Boolean, value: Any): Unit = {
+      testWithUnpredictableCasts(expected, value, classOf[Unit],
+          _.isInstanceOf[Unit], _.asInstanceOf[Unit])
+    }
+
+    @inline def testBoxedUnit(expected: Boolean, value: Any): Unit = {
+      testWithUnpredictableCasts(expected, value, classOf[BoxedUnit],
+          _.isInstanceOf[BoxedUnit], _.asInstanceOf[BoxedUnit])
+    }
+
+    testUnit(true, ())
+    testBoxedUnit(true, ())
+
+    testUnit(false, 12345)
+    testBoxedUnit(false, 12345)
+  }
+
+  @Test def string(): Unit = {
+    @inline def testString(expected: Boolean, value: Any): Unit =
+      test(expected, value, classOf[String], _.isInstanceOf[String], _.asInstanceOf[String])
+
+    testString(true, "hello")
+    testString(true, "a")
+
+    testString(false, 1)
+    testString(false, List(5))
+
+    testNullValue(classOf[String], _.isInstanceOf[String], _.asInstanceOf[String])
+  }
+
+  @Test def strictFloats(): Unit = {
+    assumeTrue("Assumed strict floats", hasStrictFloats)
+
+    @inline def testFloat(expected: Boolean, value: Any): Unit =
+      test(expected, value, classOf[Float], _.isInstanceOf[Float], _.asInstanceOf[Float])
+
+    testFloat(false, 1.2)
```

Comment:
`!executingInJVM` is used when it's true in JS and false on the JVM. Here's it's false on both.

Reply:
Ah, I see what you mean, now. Yes, it's true that it would be a bit nicer to integrate it in `primitiveTests`.

Except for the `match` test, but maybe we can just drop that one. It's a consequence of everything else, and there's no real reason that we're testing *that* case with a `match`, but none of the dozens of other cases.